std/uri.d

```diff
@@ -372,6 +372,47 @@ string encodeComponent(Char)(in Char[] uriComponent) if (isSomeChar!Char)
     return URI_Encode(s, URI_Alpha | URI_Digit | URI_Mark);
 }
 
+/* Encode associative array using www-form-urlencoding
+ *
+ * Params:
+ *      values = an associative array containing the values to be encoded.
+ *
+ * Returns:
+ *      A string encoded using www-form-urlencoding.
+ */
+package string urlEncode(int reserve = -1)(in string[string] values)
+{
+    if (values.length == 0)
+        return "";
+
+    import std.array : Appender;
+
+    Appender!string enc;
+    static if (reserve < 0)
+        enc.reserve(values.length * 128);
+    else
+        enc.reserve(reserve);
+
+    bool first = true;
+    foreach (k, v; values)
```

Comment:
more like this. I was mentioning because of AA.byKeyValue(), but that does not work because good as you would need format inside of a map which would defied the purpose of using an appender.  

auto enc = Appender!string();
foreach(k, v; values) {
    if(!first) {
       enc.put("&");
       first = false;
    }
    formattedWrite(enc, "%s=%s", encodeComponent(k), encodeComponent(v));
}